yldio/asap-hub
# CRN-1083 - Change from Contentful Schema to Migrations

# CRN-1083 - Change from Contentful Schema to Migrations

## This PR handles

- Adding `contentful-migrate`;
- All migrations to mirror what we had in schema;
- Removing the schema;
- Adding Contentful variables to `.env.example`;
- Documenting how to add and run migrations;
- Yarn convenience methods to run the migrations.
- GitHub action to run migrations on merge to `master`.

## Testing

- Set all env variables to point at the `Migration` environment (see README);
- Delete all content types from `Migration` environment on Contentful;
- Run `yarn workspace @asap-hub/contentful ctf-migrate init` to add the `Migrations` back;
- Run `yarn workspace @asap-hub/contentful space:migrate:crn` to run the migrations.

## Additionally

- Merging to master should run the migrations on `Production` and `Development`;
- I have already run `init` on these envs.
